proxy/src/telemetry/metrics/http.rs

```diff
@@ -0,0 +1,84 @@
+use std::fmt;
+
+use telemetry::metrics::*;
+
+pub(super) type RequestScopes = Scopes<RequestLabels, RequestMetrics>;
+
+#[derive(Debug, Default)]
+pub(super) struct RequestMetrics {
+    total: Counter,
+}
+
+pub(super) type ResponseScopes = Scopes<ResponseLabels, ResponseMetrics>;
+
+#[derive(Debug, Default)]
+pub struct ResponseMetrics {
+    total: Counter,
+    latency: Histogram<latency::Ms>,
+}
+
+// ===== impl RequestScopes =====
+
+impl RequestScopes {
+    metrics! {
+        request_total: Counter { "Total count of HTTP requests." }
+    }
+}
+
+impl fmt::Display for RequestScopes {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        if self.scopes.is_empty() {
+            return Ok(());
+        }
+
+        Self::request_total.fmt_help(f)?;
+        Self::request_total.fmt_scopes(f, &self, |s| &s.total)?;
+
+        Ok(())
+    }
+}
+
+// ===== impl RequestMetrics =====
+
+impl RequestMetrics {
+    pub fn end(&mut self) {
```

Comment:
I like that the increment actions are now grouped in semantically related methods. Makes `Record` much less wordy! :)